File: admin-lingopro/src/models/Topic.ts
import { Timestamp } from "firebase/firestore";

export type TopicStatus = "active" | "inactive" | "deleted";

export interface Topic {
  topicId: string;
  name: string;
  description: string;
  order: number;
  status: TopicStatus;
  createdAt: Timestamp;
  updatedAt: Timestamp;
  createdBy: string;
  updatedBy: string;
}

File: app/src/main/java/com/example/englishlearningapp/data/model/Topic.kt
package com.example.englishlearningapp.data.model

import com.google.firebase.firestore.PropertyName

data class Topic(
    @get:PropertyName("topicId")
    @set:PropertyName("topicId")
    var id: String = "",

    @get:PropertyName("topicName")
    @set:PropertyName("topicName")
    var name: String = "",

    @get:PropertyName("imageUrl")
    @set:PropertyName("imageUrl")
    var imageUrl: String = "",

    @get:PropertyName("status")
    @set:PropertyName("status")
    var status: String = "active"
) {
    constructor() : this("", "", "", "active")
}
